docs/src/threadpool_stats.rst

```diff
@@ -0,0 +1,59 @@
+
+.. _threadpool_stats:
+
+Thread pool statistics
+===========================
+
+Libuv provides a threadpool which can be used to run user code and get notified
+in the loop thread. It is also used internally by libuv.  The threadpool is
+global and shared across all event loops, see :c:ref:`threadpool` for more
+information.
+
+It is possible to request statistics on the threadpool activity to observe the
+number of idle threads, and the number of queued work items waiting for an idle
+thread to do the work. This may be used to detect performance problems, or to
+tune the static threadpool size for a specific work load.
+
+
+Data types
+----------
+
+.. c:type:: uv_queue_stats_t
+
+.. c:type:: (*uv_queue_stats_cb)(int queued, int idle_threads, void* data)
+
+    `queued` is the number of submitted items not yet started by a thread.
+    `idle_threads` is the number of threads waiting for an item to start.
+    `data` is the corresponding value from the :c:type:`uv_queue_stats_t`
+    structure.
+
+    .. warning::
+      These callback functions are not thread-safe. No libuv API functions may
+      be called in these callbacks other than :c:func:`uv_async_send()`.
```

Comment:
These two sentences sound contradictory to me – isn’t it the case that callback function *needs* to be thread-safe since it may be called from any thread?

Reply:
Agreed. The first sentence "These callback functions are not thread-safe" seems redundant and a bit confusing. I think the second sentence says it rightly.